controller/update.go

```diff
@@ -144,6 +144,8 @@ func (c controller) addFirst(ctx context.Context, req Request, opts UpdateOption
 }
 
 func (c controller) runAddWorker(ctx context.Context, req Request, opts UpdateOptions) (Request, error) {
+	c.Config.Logger.Info(nil, "controller: adding units %v", req.SliceIDs)
```

Comment:
Bah, I shouldn't respond to comments when I'm in the pub. It's very useful to have the context information in. If the issue is that logging is ugly, we can address that separately.